pkg/services/ngalert/accesscontrol/receivers.go

```diff
@@ -307,3 +322,72 @@ func (s ReceiverAccess[T]) AuthorizeUpdateByUID(ctx context.Context, user identi
 func (s ReceiverAccess[T]) AuthorizeReadSome(ctx context.Context, user identity.Requester) error {
 	return s.read.AuthorizePreConditions(ctx, user)
 }
+
+// All access permissions for a given receiver.
+
+// Access returns the permission sets for a slice of receivers. The permission set includes secrets, write, and
+// delete which corresponds the given user being able to read, write, and delete each given receiver.
+func (s ReceiverAccess[T]) Access(ctx context.Context, user identity.Requester, receivers ...T) (map[string]models.ReceiverPermissionSet, error) {
```

Comment:
I am not sure about this method because we assert permissions via evaluator, which can cause avalanche of logs (every evaluation of access control is logged). I wonder what profiler says about it. On other side, the receiver permissions are quite complex because we have to consider all possible permutations of legacy and new actions.
The alternative is not great too but it's used in other services - we just get user permissions, extract actions and scopes and create permission set.
@IevaVasiljeva what do you think about this approach (using evaluator to reconstruct user permissions for each resource)?